alicloud/resource_alicloud_hbase_instance.go

```diff
@@ -0,0 +1,310 @@
+	request.MasterInstanceType = Trim(d.Get("master_instance_type").(string))
+	request.CoreInstanceType = Trim(d.Get("core_instance_type").(string))
+
+	request.CoreInstanceQuantity = strconv.Itoa(d.Get("core_instance_quantity").(int))
+	request.CoreDiskType = Trim(d.Get("core_disk_type").(string))
+	request.CoreDiskQuantity = "4"
+	request.CoreDiskSize = strconv.Itoa(d.Get("core_disk_size").(int))
+	request.PayType = Trim(d.Get("pay_type").(string))
+	request.PricingCycle = Trim(d.Get("pricing_cycle").(string))
+	request.Duration = Trim(d.Get("duration").(string))
+	request.AutoRenew = strconv.FormatBool(d.Get("auto_renew").(bool))
+	request.NetType = string("classic")
+	vswitchId := Trim(d.Get("vswitch_id").(string))
+	if vswitchId != "" {
+		request.NetType = string("vpc")
+		request.VSwitchId = vswitchId
+		// check vswitchId in zone
+		vsw, err := vpcService.DescribeVSwitch(vswitchId)
+		if err != nil {
+			return nil, WrapError(err)
+		}
+
+		if request.ZoneId == "" {
+			request.ZoneId = vsw.ZoneId
+		} else if strings.Contains(request.ZoneId, MULTI_IZ_SYMBOL) {
+			zonestr := strings.Split(strings.SplitAfter(request.ZoneId, "(")[1], ")")[0]
+			if !strings.Contains(zonestr, string([]byte(vsw.ZoneId)[len(vsw.ZoneId)-1])) {
+				return nil, WrapError(Error("The specified vswitch %s isn't in the multi zone %s.", vsw.VSwitchId, request.ZoneId))
+			}
+		} else if request.ZoneId != vsw.ZoneId {
+			return nil, WrapError(Error("The specified vswitch %s isn't in the zone %s.", vsw.VSwitchId, request.ZoneId))
+		}
+
+		request.VpcId = vsw.VpcId
+	}
+
+	request.IsColdStorage = strconv.FormatBool(d.Get("is_cold_storage").(bool))
+
+	request.SecurityIPList = LOCAL_HOST_IP
+	if len(d.Get("security_ip_list").(*schema.Set).List()) > 0 {
+		request.SecurityIPList = strings.Join(expandStringList(d.Get("security_ip_list").(*schema.Set).List())[:], COMMA_SEPARATED)
+	}
+	return request, nil
+}
+
+func resourceAlicloudHBaseInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	hBaseService := HBaseService{client}
+
+	request, err := buildHBaseCreateRequest(d, meta)
+	if err != nil {
+		return WrapError(err)
+	}
+
+	raw, err := client.WithHbaseClient(func(client *hbase.Client) (interface{}, error) {
+		return client.CreateInstance(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_hbase_instance", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	response, _ := raw.(*hbase.CreateInstanceResponse)
+
+	d.SetId(response.ClusterId)
+
+	stateConf := BuildStateConf([]string{Hb_LAUNCHING, Hb_CREATING}, []string{Hb_ACTIVATION}, d.Timeout(schema.TimeoutCreate),
+		10*time.Minute, hBaseService.HBaseClusterStateRefreshFunc(d.Id(), []string{Hb_CREATE_FAILED}))
+	if _, err := stateConf.WaitForState(); err != nil {
+		return WrapError(err)
+	}
+
+	return resourceAlicloudHBaseInstanceUpdate(d, meta)
```

Comment:
I noticed the `name` has been set in the createInstance request and the update method only modify the name, so line 226 should return resourceAlicloudHBaseInstanceRead directly.